games-emulation/jrommanager/jrommanager-1.6-r1.ebuild

```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MY_PN=JRomManager
+MY_PV=1.6b20
+MY_P="${MY_PN}-${MY_PV}"
+
+inherit desktop
+
+DESCRIPTION="A Mame and Retrogaming Rom Manager, Java alternative to ClrMamePro"
+HOMEPAGE="https://github.com/optyfr/JRomManager"
+SRC_URI="https://github.com/optyfr/${MY_PN}/releases/download/${MY_PV}/${MY_P}.zip -> ${P}.zip"
+QA_PREBUILT="*"
+
+S="${WORKDIR}"
+
+LICENSE="GPL-3"
```

Comment:
`JRomManager/src/jrm/compressors/zipfs` is GPL-2.

Is that library used inside the main .jar? If so, 'GPLv3 software can only be combined and share code with GPLv2 software if the used GPLv2 license has the optional "or later" clause' (source: https://en.wikipedia.org/wiki/GNU_General_Public_License). In other words, this program can't be packaged within Gentoo unless upstream fixes licensing. I'll report it upstream when I get confirmation.

Besides, there are a few other .jars bundled in the binary archive, so you need to list licenses for them as well.

CC @gentoo/licenses.

Reply:
It is used. So I should be GPL-2 because it's GPL-2? Does that changes something if I bundle that part into a separate Jar? And what's about other bundled jars? they are mostly MIT, APACHE 2.0, or LGPL 2.1 ...  what should be my own licensing with all that ones?

Where should I list all licenses? in ebuild? in main jar? both? elsewhere?

Thanks for your reply

Reply:
Ah, you are the upstream. So:

1. List all licenses in ebuild's `LICENSE`.
2. The easy way out would be to relicense your code into GPL-2, so it would all be the same license.
3. As for splitting it out, I'm afraid I can't answer that question. License issues are hard, and apparently you can't link GPL-2 and GPL-3 together. However, I'm not sure how far does 'link' qualify in regards to Java. @gentoo/licenses might know more.

Reply:
Ok, looking into it more it seems that Oracle gives you [classpath exception](https://docs.oracle.com/cd/E65412_01/E71326/html/license-classpath-exception.html) that should cover this. However, you are missing the original `LICENSE` file from zipfs. Once you readd that to its directory, you should be fine with those terms. Then, just add `GPL-2-with-classpath-exception` to LICENSE along with other licenses.

Reply:
@gentoo/licenses seems to be a private member to me, hope he can answer here anyway

Reply:
Until I get clarification from @gentoo/licenses the LICENSE will be set to:
LICENSE="GPL-3 GPL-2-with-classpath-exception BSD-2 MIT Apache-2.0 LGPL-2.1"